src/elasticsearch/elasticsearch.ts

```diff
@@ -31,22 +30,58 @@ export function getShardInfo(instanceId: string): Promise<object> {
         .then(JSON.parse);
 }
 
+export function getDocuments(node: ElasticsearchNode): Promise<Documents> {
+    return ssmCommand(`curl localhost:9200/_nodes/${node.ec2Instance.privateIp}/stats`, node.ec2Instance.id)
+        .then(response => {
+            const nodeStats: NodeStats = JSON.parse(response);
+            return nodeStats.nodes[node.nodeId].indices.docs;
+        });
+}
+
+export function updateRebalancingStatus(instanceId: string, status: string): Promise<boolean> {
+    const disableRebalancingCommand: object =
+        {
+            "persistent": {
+                "cluster.routing.rebalance.enable": status
+            }
+        };
+    const elasticsearchCommand = `curl -f -v -X PUT "localhost:9200/_cluster/settings" -H 'Content-Type: application/json' -d '${JSON.stringify(disableRebalancingCommand)}'`;
+    return ssmCommand(elasticsearchCommand, instanceId)
+        .then((rawResponse: StandardOutputContent) => {
+            if (!successfulAction(rawResponse)) {
+                throw `unexpected Elasticsearch response when attempting to update re-balancing status, we got: ${rawResponse}`;
+            } else {
+                console.log(`Successfully updated re-balancing status of cluster to: ${status}`);
+                return true;
+            }
+        })
+}
+
+export function migrateShards(moves: Move[], oldestElasticsearchNode: ElasticsearchNode): Promise<boolean> {
+
+    function buildMoveCommand(move: Move): MoveCommand {
+        return { "move": move };
+    }
 
-export function migrateShards(moves: Move[], oldestElasticsearchNode: ElasticsearchNode): Promise<DefaultResponse> {
     const commands: MoveCommand[] = moves.map(buildMoveCommand);
     const rerouteCommand: RerouteCommand = {
         commands
     };
     console.log(`Re-route command will be: ${JSON.stringify(rerouteCommand)}`);
     const elasticsearchCommand = `curl -f -v -X POST "localhost:9200/_cluster/reroute" -H 'Content-Type: application/json' -d '${JSON.stringify(rerouteCommand)}'`;
     return ssmCommand(elasticsearchCommand, oldestElasticsearchNode.ec2Instance.id)
-        .then(() => {
-                const response: DefaultResponse = {"oldestElasticsearchNode": oldestElasticsearchNode};
-                return response;
+        .then((rawResponse: StandardOutputContent) => {
+            if (!successfulAction(rawResponse)) {
+                throw `unexpected Elasticsearch response when attempting to migrate shards, we got: ${rawResponse}`;
+            } else {
+                console.log(`Successfully began migrating shards off ES node: ${JSON.stringify(oldestElasticsearchNode)}`);
+                return true;
             }
-        )
+        })
+
 }
 
-function buildMoveCommand(move: Move): MoveCommand {
-    return { "move": move };
+function successfulAction(rawResponse: StandardOutputContent): Boolean {
```

Comment:
`boolean` since the casing is important 😄 

> Number, String, Boolean, and Object
> These types refer to non-primitive boxed objects that are almost never used appropriately in JavaScript code. [...] Do use the types number, string, and boolean.

Reply:
Thanks @katebee - Scala muscle memory strikes again!